connect/runtime/src/test/java/org/apache/kafka/connect/integration/DeadLetterQueueIntegrationTest.java

```diff
@@ -0,0 +1,145 @@
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+@Category(IntegrationTest.class)
+public class DeadLetterQueueIntegrationTest {
+
+    private static final Logger log = LoggerFactory.getLogger(DeadLetterQueueIntegrationTest.class);
+
+    private static final String DLQ_TOPIC = "my-connector-errors";
+
+    private final EmbeddedConnectCluster connect = new EmbeddedConnectCluster(ConnectIntegrationTest.class);
+
+    @Before
+    public void setup() throws IOException {
+        connect.start();
+    }
+
+    @After
+    public void close() {
+        connect.stop();
+    }
+
+    @Test
+    public void testTransformationErrorHandling() throws Exception {
+        // create test topic
+        connect.kafka().createTopic("test-topic");
+
+        // produce some strings into test topic
+        for (int i = 0; i < 10; i++) {
+            connect.kafka().produce("test-topic", "hello-" + i, String.valueOf(i));
+            connect.kafka().produce("test-topic", "world-" + i, String.valueOf(i));
+        }
+
+        // consume all records from test topic
+        log.info("Consuming records from test topic");
+        for (ConsumerRecord<byte[], byte[]> recs : connect.kafka().consumeNRecords(20, 5000, "test-topic")) {
+            log.debug("Consumed record ({}, {}) from topic {}", new String(recs.key()), new String(recs.value()), recs.topic());
+        }
+
+        Map<String, String> confs = new HashMap<>();
+        confs.put("connector.class", "MonitorableSink");
+        confs.put("task.max", "2");
+        confs.put("topics", "test-topic");
+        confs.put("key.converter", StringConverter.class.getName());
+        confs.put("value.converter", StringConverter.class.getName());
+
+        confs.put("errors.log.enable", "true");
+        confs.put("errors.log.include.messages", "false");
+
+        confs.put("errors.deadletterqueue.topic.name", DLQ_TOPIC);
+        confs.put("errors.deadletterqueue.topic.replication.factor", "1");
+        confs.put("errors.tolerance", "all");
+        confs.put("transforms", "failing_transform");
+        confs.put("transforms.failing_transform.type", FaultyPassthrough.class.getName());
+
+        connect.startConnector("simple-conn", confs);
+
+        int attempts = 0;
+        while (attempts++ < 5 && MonitorableSinkConnector.COUNTER.get() < 18) {
+            log.info("Received only {} records. Waiting .. ", MonitorableSinkConnector.COUNTER.get());
+            Thread.sleep(500);
+        }
+
+        Assert.assertEquals(18, MonitorableSinkConnector.COUNTER.get());
+
+        // consume failed records from dead letter queue topic
+        log.info("Consuming records from test topic");
+        for (ConsumerRecord<byte[], byte[]> recs : connect.kafka().consumeNRecords(2, 5000, DLQ_TOPIC)) {
```

Comment:
I think N in `consumeNRecords` is redundant. The fact that it accepts an argument (that ideally has a name vs a value) should be enough. `consume`?